### PR TITLE
Refresh README.md, CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by hauntsaninja**
This was prompted by #11403

Changes include:
- General streamlining of markdown
- Remove outdated references, e.g. to Python 2 / how to install Python 3
- Mention the "type school" discussions repo, typing-sig mailing list
- Remove roadmap, since there isn't really a mypy core team anymore
- More concretely link to "good first issue" page, etc. Hopefully this
    is a better experience than the somewhat common case of gitter posts
    like "I know Python, please help me contribute" that don't
    really go anywhere
- Consolidate contributing information in CONTRIBUTING.md
- Remove issue tracker conventions, since the template does a pretty
    good job and the rest is either common sense or never looked at

### CODE REVIEW

File: CREDITS [hunk 1]
```diff
@@ -10,14 +10,16 @@ the release blog posts at https://mypy-lang.blogspot.com/.
...
-  Michael J. Sullivan <sully@msully.net>
 
 Non-Dropbox core team members:
 
   Ethan Smith
+  Guido van Rossum <guido@python.org>
   Jelle Zijlstra
+  Michael J. Sullivan <sully@msully.net>
+  Shantanu Jain
+  Xuanda Yuang
```
Comment (TH3CHARLie) [inline CREDITS:22]: My name is `Xuanda Yang` : )

**Review by TH3CHARLie — APPROVED** [review]
Nice refreshment! Thanks for doing this!

File: CREDITS (hunk 1 above)
Comment (hauntsaninja) [reply CREDITS:22]:
```suggestion
  Xuanda Yang
```

Comment (hauntsaninja) [reply CREDITS]: Oh I'm so sorry!!! Fixed :-)

**Comment by pranavrajpal** [comment]
One other change that I'd like to see that's related to this is making the developer documentation more prominent. Right now, the link to the developer documentation is barely mentioned once at the bottom of CONTRIBUTING.md, which makes it very easy to not notice that it exists unless you're specifically looking for developer documentation.

I think we could improve this by moving that link to either the top of CONTRIBUTING.md or the Contributing section of the README.

**Comment by TH3CHARLie** [comment]
very reasonable, feel free to make a PR

**Pull request merged**